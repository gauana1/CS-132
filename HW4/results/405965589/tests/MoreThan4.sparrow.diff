8c8,9
< 2
---
> 5
> 4
10d10
< 3
12,13c12,13
< --- Error (Explicit) ---
< null
---
> 1
> 0
