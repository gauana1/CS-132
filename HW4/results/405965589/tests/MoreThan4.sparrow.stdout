func main()
t2 = 4
t3 = 8
t4 = alloc(t2)
t5 = alloc(t3)
t2 = @MT4_Start
[t5 + 0] =t2
t3 = @MT4_Change
[t5 + 4] =t3
[t4 + 0] =t5
if0 t4 goto lbl_NullPtr_6
goto lbl_GoodPtr_7
lbl_NullPtr_6:
error ("null pointer")
lbl_GoodPtr_7:
if0 t4 goto lbl_NullPtr_10
t2 = [t4 + 0]
t2 = [t2 + 0]
t3 = 1
t5 = 2
s1 = 3
s2 = 4
s3 = 5
s4 = 6
a6 = s2
a5 = s1
a7 = s3
a2 = t4
a4 = t5
a3 = t3
v17 = s4
s5 = call t2( v17)
goto lbl_GoodPtr_11
lbl_NullPtr_10:
error ("null pointer")
lbl_GoodPtr_11:
print (s5)
t4 = 0
v18 = t4
s5 = stack_s5
s1 = stack_s1
s3 = stack_s3
s2 = stack_s2
s4 = stack_s4
return v18
func MT4_Start(p6)
s2 = p6
stack_s5 = s5
stack_s2 = s2
stack_s4 = s4
print (a3)
print (a4)
print (a5)
print (a6)
print (a7)
print (s2)
if0 a2 goto lbl_NullPtr_21
s4 = [a2 + 0]
s4 = [s4 + 4]
stack_a8 = a3
stack_a2 = a2
stack_a3 = a3
a3 = s2
t0 = a5
stack_a5 = a5
a5 = a6
stack_a6 = a6
a6 = t0
t0 = a4
stack_a4 = a4
a4 = a7
stack_a7 = a7
a7 = t0
s5 = call s4( stack_a8)
a2 = stack_a2
a3 = stack_a3
a4 = stack_a4
a5 = stack_a5
a6 = stack_a6
a7 = stack_a7
a3 = stack_a8
goto lbl_GoodPtr_22
lbl_NullPtr_21:
error ("null pointer")
lbl_GoodPtr_22:
t2 = s5
aux = t2
s5 = stack_s5
s2 = stack_s2
s4 = stack_s4
return aux
func MT4_Change(p6)
s2 = p6
stack_s2 = s2
print (a3)
print (a4)
print (a5)
print (a6)
print (a7)
print (s2)
t2 = 0
v23 = t2
s2 = stack_s2
return v23
